walkthroughs/week-5-tdd/project/src/main/java/com/google/sps/FindMeetingQuery.java

```diff
@@ -134,6 +145,36 @@ else if ( eventStartTime == bookedTimeRange.start()){
         availableTimeRanges.add(availableTimeToBook);
     }
 
+    Collection<TimeRange> availableTimeRangesWithOptional = new ArrayList<>();
+    Set<String> optionalAttendees = new HashSet<>(request.getOptionalAttendees());
+  
+    //check the booked TimeRange of optional attendees to see if they can be fitted into the available TimeRange already derived.
```

Comment:
Anytime you feel you need to add a comment like this to explain what's going on, consider extracting that block into a helper method instead. Comments can go stale, function names are typically easier to keep in sync.